### PR TITLE
Implement query service

### PR DESCRIPTION
**Pull request by Jibandeep**
Implement query service

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,250 +1,251 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (SantoshNC68) [inline pom.xml]: Please check the pom.xml spacing, Tabs should be replaced with 4 spaces

File: src/main/java/com/ericsson/ei/queryservice/ProcessAggregatedObject.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/***********************************************************************
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/queryservice/ProcessAggregatedObject.java]: Please remove the ericsson internal headers and replace with the Ericsson Github headers in all files

File: src/main/java/com/ericsson/ei/queryservice/ProcessAggregatedObject.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+     * Aggregation database and return it as JSONArray.
+     *
+     * @param request
+     * @param AggregationDataBaseName
+     * @param AggregationCollectionName
+     * @return JSONArray
+     */
+    public JSONArray processQueryAggregatedObject(JsonNode request, String AggregationDataBaseName,
+            String AggregationCollectionName) {
+        DB db = new MongoClient("localhost", 27018).getDB(AggregationDataBaseName);
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/queryservice/ProcessAggregatedObject.java]: this db can be final,
Should we really create a connection here? Can we not have a connection class to fetch the connections?? 
Why do we need to hardcode the connection properties, is it something that is not fetched from the properties file??

File: src/main/java/com/ericsson/ei/queryservice/ProcessMissedNotification.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
+/***********************************************************************
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/queryservice/ProcessMissedNotification.java]: Ericsson Github headers should be used here

File: src/main/java/com/ericsson/ei/queryservice/ProcessMissedNotification.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+     * missed Notification database and return it as JSONArray.
+     *
+     * @param request
+     * @param MissedNotificationDataBaseName
+     * @param MissedNotificationCollectionName
+     * @return JSONArray
+     */
+    public JSONArray processQueryMissedNotification(JsonNode request, String MissedNotificationDataBaseName,
+            String MissedNotificationCollectionName) {
+        DB db = new MongoClient("localhost", 27018).getDB(MissedNotificationDataBaseName);
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/queryservice/ProcessMissedNotification.java]: Not good to have so many connections created in every class, the same code is getting repeated, move this to a common class

File: src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+	 * @throws IOException
+	 */
+	public JSONArray filterFormParam(JsonNode request) throws JsonProcessingException, IOException {
+		JsonNode criteria = request.get("criteria");
+		JsonNode options = request.get("options");
+		log.info("The criteria is : " + criteria.toString());
+		log.info("The options is : " + options.toString());
+		JSONArray resultAggregatedObject = null;
+		JSONArray resultMissedNotification = null;
+		if (options.toString().equals("{}") || options.isNull()) {
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java]: strings which are to be checked have to be the first ones, ex: if("".equals(obj))

File: src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+	 * process it to create a JsonNode request to query the Aggregated Objects.
+	 * If the Aggregated Object matches the condition, then it is returned.
+	 *
+	 * @param request
+	 * @return JSONArray
+	 * @throws JsonProcessingException
+	 * @throws IOException
+	 */
+	public JSONArray filterFormParam(JsonNode request) throws JsonProcessingException, IOException {
+		JsonNode criteria = request.get("criteria");
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java]: the values which are not altered can be final variables

File: src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+	 * Aggregated Objects. If the Aggregated Object matches the condition, then
+	 * it is returned.
+	 *
+	 * @param request
+	 * @return JSONArray
+	 * @throws JsonProcessingException
+	 * @throws IOException
+	 */
+	public JSONArray filterQueryParam(HttpServletRequest request) throws JsonProcessingException, IOException {
+		String url = request.getRequestURI();
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java]: Please check the variables, if they are not altered can be declared final

File: src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+		log.info("The URI is :" + url);
+		String queryString = request.getQueryString();
+		log.info("The query string is : " + queryString);
+		ObjectMapper mapper = new ObjectMapper();
+		ObjectNode criteria = mapper.createObjectNode();
+		String[] criterias = queryString.split(",");
+		log.info("The query parameters are :");
+		for (String s : criterias) {
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java]: Better naming than "s"

Comment (SantoshNC68) [reply src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java]: Please dont mention any duraci in the pull requests in Github

**Comment by SantoshNC68** [comment]
Please dont mention duraci numbers with the pull requests

File: src/main/java/com/ericsson/ei/queryservice/ProcessMissedNotification.java (hunk 5 above)
Comment (evasiba-ericsson) [inline src/main/java/com/ericsson/ei/queryservice/ProcessMissedNotification.java]: The same comment about Mongodbhandler as I wrote above.

The fields and arguments to methods need to start with lower case letter.

File: src/test/java/com/ericsson/ei/queryservice/test/MongoHarness.java [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+	static String host = "localhost";
+	static int port = 27018;
+
+	static MongoCollection<Document> col;
+
+	private static MongodExecutable mongodExecutable;
+
+	@BeforeClass
+	public static void beforeClass() throws IOException {
```
Comment (evasiba-ericsson) [inline src/test/java/com/ericsson/ei/queryservice/test/MongoHarness.java]: Change the set up of embedded mongo db to the one we use in test case for mongo DB handler. That one chooses available ports.

**Pull request closed without merging**